lms/djangoapps/course_structure_api/v0/api.py

```diff
@@ -0,0 +1,94 @@
+"""
+API implementation of the Course Structure API for Python code.
+
+Note: The course list and course detail functionality isn't currently supported here because of the tricky interactions between DRF and the code.
+Most of that information is available by accessing the course objects directly.
+"""
+
+from course_structure_api.v0 import serializers
+from course_structure_api.v0.errors import CourseNotFoundError, CourseStructureNotAvailableError
+from openedx.core.djangoapps.content.course_structures import models, tasks
+from courseware import courses
+
+
+def _retrieve_course(course_key):
+    """Retrieves the course for the given course key.
+
+    Args:
+        course_key: The CourseKey for the course we'd like to retrieve.
+    Returns:
+        the course that matches the CourseKey
+    Raises:
+        CourseNotFoundError
+
+    """
+    try:
+        course = courses.get_course(course_key)
+        return course
+    except ValueError:
+        raise CourseNotFoundError
+
+
+def course_structure(course_key):
+    """
+    Retrieves the entire course structure, including information about all the blocks used in the course.
+
+    Args:
+        course_key: the CourseKey of the course we'd like to retrieve.
+    Returns:
+        The serialized output of the course structure:
+            * root: The ID of the root node of the course structure.
+
+            * blocks: A dictionary that maps block IDs to a collection of
+            information about each block. Each block contains the following
+            fields.
+
+            * id: The ID of the block.
```

Comment:
I think these fields should be indented four spaces to visually indicate that they belong to the blocks dict. See the account API for an example.